tdrs-backend/tdpservice/parsers/test/test_validators.py

```diff
@@ -1229,3 +1233,120 @@ def test_fam_affil_fed_disability(self, record):
+        """Test add_record logic."""
+        case_consistency_validator = CaseConsistencyValidator(small_correct_file_header,
+                                                              util.make_generate_parser_error(small_correct_file, None))
+
+        for record, schema in zip(tanf_s1_records, tanf_s1_schemas):
+            case_consistency_validator.add_record(record, schema, True)
+
+        assert case_consistency_validator.has_validated is False
+        assert case_consistency_validator.case_has_errors is True
+        assert len(case_consistency_validator.records) == 4
+        assert case_consistency_validator.total_cases_cached == 0
+        assert case_consistency_validator.total_cases_validated == 0
+
+        # Add record with different case number to proc validation again and start caching a new case.
+        t1 = TanfT1Factory.create()
+        t1.CASE_NUMBER = 2
+        case_consistency_validator.add_record(t1, tanf_s1_schemas[0], False)
+        assert case_consistency_validator.has_validated is False
+        assert case_consistency_validator.case_has_errors is False
+        assert len(case_consistency_validator.records) == 1
+        assert case_consistency_validator.total_cases_cached == 1
+        assert case_consistency_validator.total_cases_validated == 0
+
+        # Complete the case to proc validation and verify that it occured. Even if the next case has errors.
+        t2 = TanfT2Factory.create()
+        t3 = TanfT3Factory.create()
+        t2.CASE_NUMBER = 2
+        t3.CASE_NUMBER = 2
+        case_consistency_validator.add_record(t2, tanf_s1_schemas[1], False)
+        case_consistency_validator.add_record(t3, tanf_s1_schemas[2], False)
+        assert case_consistency_validator.case_has_errors is False
+
+        case_consistency_validator.add_record(tanf_s1_records[0], tanf_s1_schemas[0], True)
+
+        assert case_consistency_validator.has_validated is True
+        assert case_consistency_validator.case_has_errors is True
+        assert len(case_consistency_validator.records) == 1
+        assert case_consistency_validator.total_cases_cached == 2
+        assert case_consistency_validator.total_cases_validated == 1
+
+    @pytest.mark.django_db
+    def test_section1_fail(self, small_correct_file_header, small_correct_file, tanf_s1_records, tanf_s1_schemas):
+        """Test TANF Section 1 records RPT_MONTH_YEAR don't align with header year and quarter."""
+        case_consistency_validator = CaseConsistencyValidator(small_correct_file_header,
+                                                              util.make_generate_parser_error(small_correct_file, None))
+
+        for record, schema in zip(tanf_s1_records, tanf_s1_schemas):
+            case_consistency_validator.add_record(record, schema, False)
+
+        num_errors = case_consistency_validator.validate()
+
+        assert 4 == num_errors
```

Comment:
can you add a call to `get_generated_errors` and some assertions on the messages to the test?